src/spark/sparkwallet.cpp

```
@@ -742,7 +742,12 @@ std::vector<CRecipient> CSparkWallet::CreateSparkMintRecipients(
         script.insert(script.end(), serializedCoins[i].begin(), serializedCoins[i].end());
         unsigned char network = spark::GetNetworkType();
         std::string addr = outputs[i].address.encode(network);
-        CRecipient recipient = {script, CAmount(outputs[i].v), false, addr};
+        std::string memo = outputs[i].memo;
+        const std::size_t max_memo_size = outputs[i].address.get_params()->get_memo_bytes();
+        if (memo.length() > max_memo_size) {
+            throw std::runtime_error(strprintf("Memo exceeds maximum length of %d bytes", max_memo_size));
+        }
+        CRecipient recipient = {script, CAmount(outputs[i].v), false, addr, memo};
```

cppcheck

[performance] Searching before insertion is not necessary. Instead of 'mapKeyMetadata[keyid]=CKeyMetadata(keypool.nTime)' consider using 'mapKeyMetadata.try_emplace(keyid, CKeyMetadata(keypool.nTime));'.

(stlFindInsert)